airflow/providers/tableau/operators/tableau.py

```diff
@@ -116,6 +116,8 @@ def execute(self, context: dict) -> str:
 
             response = method(resource_id)
 
+        job_id = ''
```

Comment:
You can just move the assignent of response.id outside of the if.
There is no need for assigning empty string.